src/ws/cockpitwsinstancecert.c

```diff
@@ -86,7 +86,12 @@ get_ws_https_instance (void)
   regfree (&preg);
   if (r != 0)
     {
-      warnx ("Not running in a template cgroup, unable to parse systemd unit instance.\n\n/proc/self/cgroups content follows:\n%s\n", buf);
+      /* It's expected that this function will often be called even when
+       * the client didn't send a certificate, so we shouldn't log about
+       * that.  It might be useful for debugging, though.
+       */
+
+      // warnx ("Not running in a template cgroup, unable to parse systemd unit instance.\n\n/proc/self/cgroups content follows:\n%s\n", buf);
```

Comment:
Is that for unit testing? Because in production, both the PAM module and cockpit-ws surely run in a cgroup.

Reply:
I noticed this during testing, yes, but in production there will be some cockpit-ws instances which are not running in a @[clientcert] cgroup.  The static http one comes to mind.